### PR TITLE
Remove old msc ver. Freecad 0.20 require 64 bit and windows 7 or higher. For this reason, msc libraries less than 1800 will no longer be used.

### PR DESCRIPTION
**Pull request by mosfet80**
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7036">pipeline status</a> for feature branch [PR_7036](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7036). Pipeline [566207706](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/566207706) was triggered at [aba0045](https://github.com/FreeCAD/FreeCAD/pull/7036/commits/aba00455b18084883e1624e39a46c2c73324ed73). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by donovaly** [comment]
Thanks for this PR.

However, despite we don't offer 32bit Windows builds anymore, some users might want to build this on their own. Therefore I would not remove this possibility from the code. Before this is done, a discussion should take place if everybody agrees to this.

Can you therefore please rebase you PR and only commit the MSC changes?

**Comment by yorikvanhavre** [comment]
I agree with @donovaly ... for me having the code as widely compatible as possible is more important than having clean code. There might be people using the FreeCAD codebase out there in ways we don't imagine (for ex on embedded systems, etc), and itws good to keep things as open and easy to port as possible

**Comment by donovaly** [comment]
I would merge the PR but it contains now 7 commits and it is hard to see what was actually changed.

@mosfet80 when you update your branch do this:
- pull from master int your branch
- in the log select the merge change and your change and combine them to one commit
- force-push the resulting single commit

If that does not work your you, then rebase your branch on master and force-push.
People in the developer's forum can help you with this because the optimal strategy depends on the Git tools you use.

**Comment by donovaly** [comment]
@mosfet80 , have you seen my previous comment? Now you have 10 commits in this PR. Please avoid this otherwise we cannot merge the PR easily.

File: src/Base/MemDebug.cpp [hunk 1]
```diff
@@ -128,7 +127,12 @@ MemDebug::MemDebug()
      throw std::runtime_error("Base::MemDebug::MemDebug():41: File IO Error. Can't open log file...");
    _strtime_s( timeStr, 15 );
    _strdate_s( dateStr, 15 );
-
+   logFile = fopen( "MemLog.txt", "w" );
```
Comment (donovaly) [inline src/Base/MemDebug.cpp:130]: Why this addition?

File: src/Mod/Drawing/Gui/TaskOrthoViews.cpp [hunk 2]
```diff
@@ -252,7 +250,6 @@ void orthoview::set_projection(const gp_Ax2& cs)
 
     calcCentre();
 
-    //this_view->Direction.setValue(Z_dir.X(), Z_dir.Y(), Z_dir.Z());
```
Comment (donovaly) [inline src/Mod/Drawing/Gui/TaskOrthoViews.cpp:260]: In principal OK, but has nothing to do with msc.

Please only make changes related to MSC in this PR.

File: src/Gui/PreferencePackTemplates/View.cfg [hunk 3]
```diff
@@ -27,7 +27,7 @@
           <FCInt Name="RenderCache" Value="0"/>
           <FCInt Name="RotationMode" Value="1"/>
           <FCInt Name="SegmentsPerGeometry" Value="50"/>
-          <FCBool Name="ShowAxisCross" Value="0"/>
+          <FCBool Name="ShowAxisCross" Value="0/>
```
Comment (donovaly) [inline src/Gui/PreferencePackTemplates/View.cfg:30]: This would introduce a bug

File: src/FCConfig.h [hunk 4]
```diff
@@ -38,83 +38,45 @@
...
-#   endif
+#  	ifndef FC_OS_BSD
+#  		define FC_OS_BSD
+#	endif
 #elif defined(__CYGWIN__)
 #   ifndef FC_OS_CYGWIN
-#   define FC_OS_CYGWIN
+#   	define FC_OS_CYGWIN
 // Avoid conflicts with Inventor
-#   define HAVE_INT8_T
```
Comment (donovaly) [inline src/FCConfig.h:85]: These must stay. You added them later but with wrong indentation

File: src/FCConfig.h [hunk 5]
```diff
@@ -38,83 +38,45 @@
...
-//# define HAVE_UINTPTR_T
+#   	define FC_OS_WIN64
 #   endif
+
 #elif defined(__MWERKS__) && defined(__INTEL__)
-#   ifndef FC_OS_WIN32
-#   define FC_OS_WIN32
-#   endif
+#  	ifndef FC_OS_WIN32
+#  		define FC_OS_WIN32
```
Comment (donovaly) [inline src/FCConfig.h:49]: Indentation seems wrong. In the change above it is correct.

File: src/Base/StackWalker.h [hunk 6]
```diff
@@ -157,6 +157,7 @@ class BaseExport StackWalker
 // the one which was installed by the VS-IDE)
 
 
+
```
Comment (donovaly) [inline src/Base/StackWalker.h:160]: not necessary for the PR

File: src/Base/StackWalker.cpp [hunk 7]
```diff
@@ -94,9 +94,7 @@
 
 // If VC7 and later, then use the shipped 'dbghelp.h'-file
 #pragma pack(push,8)
-
```
Comment (donovaly) [inline src/Base/StackWalker.cpp]: not necessary for the PR

File: src/Base/MemDebug.cpp [hunk 8]
```diff
@@ -144,8 +148,6 @@ MemDebug::MemDebug()
 MemDebug::~MemDebug()
 {
   _CrtMemDumpAllObjectsSince( NULL );
-  //_CrtCheckMemory( );
```
Comment (donovaly) [inline src/Base/MemDebug.cpp:153]: No change in this file is necessary for the PR

File: src/Base/Builder3D.cpp [hunk 9]
```diff
@@ -183,9 +183,6 @@ void Builder3D::addSingleLine(const Vector3f& pt1, const Vector3f& pt2, short li
 {
   char lp[20];
   sprintf(lp, "0x%x", linePattern);
-  //char lp[20] = "0x";
```
Comment (donovaly) [inline src/Base/Builder3D.cpp:189]: not necessary for the PR

File: src/App/ExpressionParser.tab.c [hunk 10]
```diff
@@ -220,8 +220,7 @@ typedef short int yytype_int16;
 # define YY_ATTRIBUTE_UNUSED YY_ATTRIBUTE ((__unused__))
 #endif
 
-#if !defined _Noreturn \
-     && (!defined __STDC_VERSION__ || __STDC_VERSION__ < 201112)
+#if !defined _Noreturn && (!defined __STDC_VERSION__ || __STDC_VERSION__ < 201112)
```
Comment (donovaly) [inline src/App/ExpressionParser.tab.c:223]: not necessary for the PR

**Review by donovaly — CHANGES_REQUESTED** [review]
Please only make changes that remove the MSC code. This assures a clean commit history. Further changes can be done in a separate PR.

**Comment by donovaly** [comment]
@mosfet80 , can you please stop committing before you read my posts and reviews?

**Pull request merged**